### ISSUE TITLE
hiprtcCompileProgram can not recognise the -I option

### ISSUE THREAD

**Issue by ipe-zhangyz** [ROCm-Developer-Tools/HIP#2182]
I tested the saxpy.cpp from the tests of hip. it works  when in its original form. However, if the "saxpy.cu" has some included headers and there need an included path option, the hiprtcCompileProgram will fail to compile the cu source program. The code is like this:

```
static constexpr auto saxpy{
R"(
#include "test.h"
#include <hip/hip_runtime.h>
extern "C"
__global__
void saxpy(float a, float* x, float* y, float* out, size_t n)
{
    size_t tid = blockIdx.x * blockDim.x + threadIdx.x;
    if (tid < n) {
       out[tid] = a * x[tid] + y[tid];
    }
}
)"};
......
    hiprtcProgram prog;
    hiprtcCreateProgram(&prog,      // prog
                        saxpy,      // buffer
                        "saxpy.cu", // name
                        0,          // numHeaders
                        nullptr,    // headers
                        nullptr);   // includeNames
    hipDeviceProp_t props;
    int device = 0;
    hipGetDeviceProperties(&props, device);
    std::string sarg = "--gpu-architecture=gfx906";
    std::string sarg1 = "-I.";

    const char* options[2];
    options[1] = sarg.c_str();
    options[0] = sarg1.c_str();
    for (int i = 0; i < 2; ++i)
        printf("option %d is %s\n", i, options[i]);
    hiprtcResult compileResult{hiprtcCompileProgram(prog, 2, options)};
```

The included header _test.h_ is in the current directory, and there is a _-I._ option in the _options_ parameter of hiprtcCompileProgram. The saxpy.cpp can be compiled normally using hipcc， but the generated executable fails when running, with these output:

> /tmp/comgr-77d2a4/input/CompileSource:2:10: fatal error: 'test.h' file not found
> #include "test.h"
>          ^~~~~~~~
> 1 error generated when compiling for gfx906.
> Error: Failed to compile opencl source (from CL or HIP source to LLVM IR).
> 
> error: Compilation failed.
> error: TEST FAILED
> Aborted

Passing the absolute path of the _test.h_ file to the _-I_ option makes no sense, while adding the absolute path of the _test.h_ file to the _#include_ line in the cu source makes the running passed. So it seems that hiprtcCompileProgram cannot recognise the _-I_ parameters in the JIT compiling process.
The test was conducted using the  rocm/dev-ubuntu-16.04:3.5 docker image from docker hub.

**Comment by cjatin** [comment]
3.5 is pretty old, can you try it with a newer version.
Also can you run the exe with AMD_COMGR_EMIT_VERBOSE_LOGS=1 env variable and share the output.

**Comment by XuanBaby** [comment]
@cjatin  I  test the saxpy.cpp and  run the exe with AMD_COMGR_EMIT_VERBOSE_LOGS=1 in docker images:rocm/dev-ubuntu-18.04:3.8，
the output is  as follow :
`option 0 is -I.
option 1 is --gpu-architecture=gfx906
amd_comgr_do_action:
	  ActionKind: AMD_COMGR_ACTION_COMPILE_SOURCE_TO_BC
	     IsaName: amdgcn-amd-amdhsa--gfx906+sram-ecc
	     Options: "-O3" "-D__HIP_ROCclr__=1" "-D__HIP_ARCH_GFX906__=1" "-mllvm" "-amdgpu-prelink" "-mcumode" "-mwavefrontsize64"
	        Path: 
	    Language: AMD_COMGR_LANGUAGE_HIP
COMGR::InProcessDriver::Execute argv: clang "-cc1" "-triple" "amdgcn-amd-amdhsa" "-aux-triple" "x86_64-unknown-linux-gnu" "-emit-llvm-bc" "-emit-llvm-uselists
" "-disable-llvm-verifier" "-discard-value-names" "-main-file-name" "CompileSource" "-mrelocation-model" "pic" "-pic-level" "1" "-mthread-model" "posix" "-mframe-pointer=none" "-fno-rounding-math" "-mconstructor-aliases" "-aux-target-cpu" "x86-64" "-fcuda-is-device" "-mllvm" "-amdgpu-internalize-symbols" "-fcuda-allow-variadic-functions" "-fvisibility" "hidden" "-fapply-global-visibility-to-externs" "-target-cpu" "gfx906" "-target-feature" "-wavefrontsize16" "-target-feature" "-wavefrontsize32" "-target-feature" "+wavefrontsize64" "-target-feature" "-xnack" "-target-feature" "+sram-ecc" "-target-feature" "+cumode" "-dwarf-column-info" "-fno-split-dwarf-inlining" "-debugger-tuning=gdb" "-resource-dir" "lib/clang/11.0.0" "-isystem" "/opt/rocm/hip/include" "-isystem" "/opt/rocm/llvm/lib/clang/11.0.0" "-isystem" "/opt/rocm/llvm/lib/clang/11.0.0/include" "-I" "/tmp/comgr-3f4ad6/include" "-D" "__HIP_ROCclr__=1" "-D" "__HIP_ARCH_GFX906__=1" "-internal-isystem" "/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0" "-internal-isystem" "/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/x86_64-linux-gnu/c++/7.5.0" "-internal-isystem" "/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/x86_64-linux-gnu/c++/7.5.0" "-internal-isystem" "/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/backward" "-internal-isystem" "/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0" "-internal-isystem" "/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/x86_64-linux-gnu/c++/7.5.0" "-internal-isystem" "/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/x86_64-linux-gnu/c++/7.5.0" "-internal-isystem" "/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/backward" "-internal-isystem" "/usr/local/include" "-internal-isystem" "lib/clang/11.0.0/include" "-internal-externc-isystem" "/usr/include/x86_64-linux-gnu" "-internal-externc-isystem" "/include" "-internal-externc-isystem" "/usr/include" "-internal-isystem" "/usr/local/include" "-internal-isystem" "lib/clang/11.0.0/include" "-internal-externc-isystem" "/usr/include/x86_64-linux-gnu" "-internal-externc-isystem" "/include" "-internal-externc-isystem" "/usr/include" "-O3" "-std=c++11" "-fdeprecated-macro" "-fno-autolink" "-fdebug-compilation-dir" "/data_share/xuan/hiprtc" "-ferror-limit" "19" "-fgnuc-version=4.2.1" "-fcxx-exceptions" "-fexceptions" "-vectorize-loops" "-vectorize-slp" "-mllvm" "-amdgpu-prelink" "-fcuda-allow-variadic-functions" "-faddrsig" "-o" "/tmp/comgr-3f4ad6/output/CompileSource.bc" "-x" "hip" "/tmp/comgr-3f4ad6/input/CompileSource"/tmp/comgr-3f4ad6/input/CompileSource:2:10: fatal error: 'test.h' file not found
#include "test.h"
         ^~~~~~~~
1 error generated when compiling for gfx906.
	ReturnStatus: AMD_COMGR_STATUS_ERROR

Error: Failed to compile opencl source (from CL or HIP source to LLVM IR).

error: Compilation failed.
error: TEST FAILED
Aborted (core dumped)`

**Comment by leofang** [comment]
Hi, I am a CuPy developer, and I can confirm as of ROCm 3.8.0 hipRTC still can't recognize the `-I` option to search headers in non-default locations. This has been reported to AMD during a PFN-AMD meeting (which I didn't attend as I'm not affiliated with either party) and hopefully it is tracked somewhere. (We had to introduce an awful workaround to recursively parse all headers and prepend them, which is a very fragile solution and worsens the already-slow performance of hipRTC.)

**Comment by leofang** [comment]
> I can confirm as of ROCm 3.8.0 hipRTC still can't recognize the `-I` option to search headers in non-default locations.

Got access to a 3.9.0 environment by accident, and confirmed it's still not fixed.

**Comment by cjatin** [comment]
Thanks for all the information.
Will keep this open while we plan a fix for it.

**Comment by leofang** [comment]
Looks like this is fixed in ROCm 4.2.0!

**Comment by SarbojitAMD** [comment]
closing the ticket, feel free to reopen incase you still see the issue.
Thanks
